### DISCUSSION TITLE
Dependency Injection - IWorkflowRunner

### DISCUSSION THREAD

**Discussion by vinujosey** [elsa-workflows/elsa-core#5661] [Q&A]
How can I inject IWorkflowRunner to one of my custom class, so that I can invoke the workflow based on user request

**Comment by glime-ai** [comment]
To inject `IWorkflowRunner` into your custom class, you can follow the standard dependency injection pattern used in .NET Core applications. Here's how you can do it:

1. Ensure that the `Elsa` services are registered in the `Startup.cs` or `Program.cs` file of your application. This is typically done by calling `AddElsa` on the `IServiceCollection` instance.

2. Define a constructor in your custom class that accepts `IWorkflowRunner` as a parameter. The dependency injection container will automatically provide an instance of `IWorkflowRunner` when creating an instance of your class.

Here's an example of what your custom class might look like:

```csharp
using Elsa.Services;

public class MyCustomClass
{
    private readonly IWorkflowRunner _workflowRunner;

    public MyCustomClass(IWorkflowRunner workflowRunner)
    {
        _workflowRunner = workflowRunner;
    }

    public async Task InvokeWorkflowAsync()
    {
        // Use the _workflowRunner to start a workflow
        var result = await _workflowRunner.RunWorkflowAsync("<WorkflowDefinitionId>");
        // Handle the result as needed
    }
}
```

And in your `Startup.cs` or `Program.cs`, make sure to register your custom class:

```csharp
services.AddTransient<MyCustomClass>();
```

Now, whenever `MyCustomClass` is resolved from the service container, it will be provided with an instance of `IWorkflowRunner`.

Remember to replace `<WorkflowDefinitionId>` with the actual ID of the workflow you want to invoke.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
